### PR TITLE
Always use skinny arrow for methods, add `gulp lint`

### PR DESCRIPTION
**Pull request by havocp**
This is a more thorough version of a commit that's in #3099  (https://github.com/bokeh/bokeh/commit/4e03d100ddc3b814e77d3fdbf210a92f0c224608). That commit is broken, too, and isn't related to #3099.

Rationale in the commit message https://github.com/bokeh/bokeh/commit/920d9af9ccd969b5d204936d375de345c2044d88

### CODE REVIEW

**Comment by havocp** [comment]
Integration test failures here look real, have to look into them.

**Comment by bryevdv** [comment]
It looks like the same error is causing most examples to fail, it is failing on this line in `_active_change`:

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/common/tool_manager.coffee#L117

And there cause is because `event_type = tool.get('event_type')` returns `undefined`

**Comment by bryevdv** [comment]
More info:  The problem starts here in `_init_tools`:
```
if et not in ['pinch', 'scroll']
        gestures[et].tools[0].set('active', true)
```
At this point, `gestures[et].tools[0]` is `ResizeTool` which is (correctly) a gesture. This `set` causes BB to return the callback here:
```
@listenTo(tool, 'change:active', () => @_active_change(tool))
```
But in this callback, `tool` is the `PreviewSaveTool`. Some JS scope weirdness causing the value of `tool` to be permanently (mistakenly) bound to `PreviewSaveTool`?

**Comment by bryevdv** [comment]
As a demonstration, this patch "fixes" the problem:
```
diff --git a/bokehjs/src/coffee/common/tool_manager.coffee b/bokehjs/src/coffee/common/tool_manager.coffee
index f9623c5..a92fb97 100644
--- a/bokehjs/src/coffee/common/tool_manager.coffee
+++ b/bokehjs/src/coffee/common/tool_manager.coffee
@@ -107,7 +107,8 @@ class ToolManager extends HasProperties
         continue
       gestures[et].tools = _.sortBy(tools, (tool) -> tool.get('default_order'))
       if et not in ['pinch', 'scroll']
-        gestures[et].tools[0].set('active', true)
+        gestures[et].tools[0].set('active', true, {silent: true})
+        @_active_change(gestures[et].tools[0])
```
There is definitely something about the way `_active_change` is bound that seems to be the problem.

**Comment by bryevdv** [comment]
This diff also fixes the problem, and is probably better, but I don't know if it is "correct"
```
diff --git a/bokehjs/src/coffee/common/tool_manager.coffee b/bokehjs/src/coffee/common/tool_manager.coffee
index f9623c5..31d6101 100644
--- a/bokehjs/src/coffee/common/tool_manager.coffee
+++ b/bokehjs/src/coffee/common/tool_manager.coffee
@@ -99,7 +99,7 @@ class ToolManager extends HasProperties
           continue
 
         gestures[et].tools.push(tool)
-        @listenTo(tool, 'change:active', () => @_active_change(tool))
+        @listenTo(tool, 'change:active', () => _.bind(@_active_change, tool))
 
     for et of gestures
       tools = gestures[et].tools
```
I don't offhand understand the exact problem, but hope this is useful info.

**Comment by bryevdv** [comment]
Of course, I was preliminary. The above patch allows the plot to be drawn at all, but now the tool manager is broken, it does not enforce only one active tool per gesture type anymore, and switching to a new active tool does not work.

**Comment by bryevdv** [comment]
Final data point, messing with the toolbar buttons at one point:
```
PlotView.prototype.request_render = function() {
            if (!this.is_paused) {
              this.throttled_render(true);
            }
      };
```
gets called with `PolySelectionView` as the value of `this`

File: bokehjs/src/coffee/common/tool_manager.coffee [hunk 1]
```diff
@@ -99,7 +99,7 @@ class ToolManager extends HasProperties
           continue
 
         gestures[et].tools.push(tool)
-        @listenTo(tool, 'change:active', _.bind(@_active_change, tool))
+        @listenTo(tool, 'change:active', () => @_active_change(tool))
```
Comment (havocp) [inline bokehjs/src/coffee/common/tool_manager.coffee]: the scope of tool isn't what we want here I guess. Putting _.bind back (with the lambda taking a tool param) would be a possible clunky fix maybe.

Comment (bryevdv) [inline bokehjs/src/coffee/common/tool_manager.coffee]: That only partially fixes things, though. The plot at least renderers, but the tool manager and event dispatching are completely broken.

**Comment by bryevdv** [comment]
Cancelled the build to save Travis resources. Running `color_scatter.py` there is some improvement but things are still quite broken with some tools not working at all and some switching between tools not working.

**Comment by havocp** [comment]
ok. I see another problem here that I'm working on.

**Comment by havocp** [comment]
Something is still messed up about switching between tools I think.

**Comment by bryevdv** [comment]
@havocp I want to add the linter and get to consistent good practice around these, but I'd like to pause this until after `0.11`given how disruptive it seems. Thoughts?

**Comment by havocp** [comment]
makes sense.

**Pull request closed without merging**